src/LFHCAL_geo.cpp

```diff
@@ -0,0 +1,1084 @@
+  DetElement det(detName, detID);
+
+  Assembly assembly(detName);
+  PlacedVolume phv;
+
+  int moduleIDx = -1;
+  int moduleIDy = -1;
+
+
+  int    moduleID   = 0;
+
+
+
+
+  std::vector<double> xpos8M;
+  std::vector<double> ypos8M;
+  std::vector<double> zpos8M;
+
+  xml_coll_t eightMPos(detElem, _Unicode(eightmodulepositions));
+  for (xml_coll_t position_i(eightMPos, _U(position)); position_i; ++position_i){
+    xml_comp_t position_comp = position_i;
+    xpos8M.push_back((position_comp.x()));
+    ypos8M.push_back((position_comp.y()));
+    zpos8M.push_back((position_comp.z()));
+  }
+
+  // create 8M modules
+  Volume  eightMassembly = createEightMModule ( desc, eightM_params, slice_Params, length, sens, renderComponents, allSensitive);
+  if (xpos8M.size() != ypos8M.size() || xpos8M.size() != zpos8M.size()){
+    std::cout << xpos8M.size() << "\t" <<ypos8M.size() <<  "\t" << zpos8M.size() << std::endl;
+    std::cout <<  "idiot you can't count" << std::endl;
+  } else {
+    for (int e = 0; e < (int)xpos8M.size(); e++){
+      if(e%20 == 0 ) std::cout <<  "LFHCAL placing 8M module: " << e << "/"<< (int)xpos8M.size() << "\t" << xpos8M[e] << "\t" << ypos8M[e] << "\t" << zpos8M[e]<< std::endl;
+        if(moduleIDx<0 || moduleIDy<0){
+        std::cout << "LFHCAL WRONG ID FOR 8M module: " << e << "/" << (int)xpos8M.size() << "\t" << moduleIDx << "\t"
+                  << moduleIDy << std::endl;
+      }
+      moduleIDx             = ((xpos8M[e] + 270) / 10);
+      moduleIDy             = ((ypos8M[e] + 265) / 10);
+
+      // Placing modules in world volume
+      auto tr8M = Transform3D(Position(pos.x()-xpos8M[e]-0.5*eightM_params.mod_width, pos.y()-ypos8M[e], pos.z() +zpos8M[e] + length / 2.));
+      phv = assembly.placeVolume(eightMassembly, tr8M);
+      phv.addPhysVolID("moduleIDx", moduleIDx).addPhysVolID("moduleIDy", moduleIDy).addPhysVolID("moduletype", 0);
+      moduleID++;
+    }
+  }
+
+  std::vector<double> xpos4M;
+  std::vector<double> ypos4M;
+  std::vector<double> zpos4M;
+
+  xml_coll_t fourMPos(detElem, _Unicode(fourmodulepositions));
+  for (xml_coll_t position_i(fourMPos, _U(position)); position_i; ++position_i){
+    xml_comp_t position_comp = position_i;
+    xpos4M.push_back((position_comp.x()));
+    ypos4M.push_back((position_comp.y()));
+    zpos4M.push_back((position_comp.z()));
+  }
+
+  // create 4M modules
+  Volume  fourMassembly = createFourMModule ( desc, fourM_params, slice_Params,  length, sens, renderComponents, allSensitive);
+  if (xpos4M.size() != ypos4M.size() || xpos4M.size() != zpos4M.size()){
+    std::cout << xpos4M.size() << "\t" <<ypos4M.size() <<  "\t" << zpos4M.size() << std::endl;
+    std::cout <<  "idiot you can't count" << std::endl;
+  } else {
+    for (int f = 0; f < (int)xpos4M.size(); f++){
+      if(f%20 == 0 )std::cout <<  "LFHCAL placing 4M module: " << f << "/"<< (int)xpos4M.size() << "\t" << xpos4M[f] << "\t" << ypos4M[f] << "\t" << zpos4M[f]<< std::endl;
```

Comment:
```suggestion
```

Reply:
Replaced all `std::cout` instances with `printout(DEBUG)` calls. Users can run with `ddsim -v DEBUG` to see them.